src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -1562,11 +1562,14 @@ public boolean hasStaticMethod(final DetailAST ident) {
         public boolean hasFinalField(final DetailAST instanceMember) {
             boolean result = false;
             for (DetailAST member : instanceMembers) {
-                final DetailAST mods = member.getParent().findFirstToken(TokenTypes.MODIFIERS);
-                final boolean finalMod = mods.findFirstToken(TokenTypes.FINAL) != null;
-                if (finalMod && isAstSimilar(member, instanceMember)) {
-                    result = true;
-                    break;
+                final DetailAST parent = member.getParent();
+                if (parent.getType() != TokenTypes.RECORD_COMPONENT_DEF) {
```

Comment:
I wasn't fully sure, but I assume this `RECORD_COMPONENT_DEF` is not a final field by default?

Reply:
record components are implicitly  `final`.

Reply:
Let me know if this code is not correct then. This is basically saying these things aren't final.

Reply:
The code should be "fixed" now.